src/io.ts

```diff
@@ -113,19 +129,21 @@ export interface AoIORead {
   }: {
     address: WalletAddress;
   }): Promise<AoGateway | undefined>;
-  getGateways(): Promise<
-    Record<WalletAddress, AoGateway> | Record<string, never>
-  >;
+  getGateways(
+    params?: PaginationParams,
+  ): Promise<PaginationResult<AoGateway & { gatewayAddress: WalletAddress }>>;
   getBalance(params: { address: WalletAddress }): Promise<number>;
-  getBalances(): Promise<Record<WalletAddress, number> | Record<string, never>>;
+  getBalances(
+    params?: PaginationParams,
+  ): Promise<PaginationResult<{ address: WalletAddress; balance: number }>>;
   getArNSRecord({
     name,
   }: {
     name: string;
   }): Promise<AoArNSNameData | undefined>;
-  getArNSRecords(): Promise<
-    Record<string, AoArNSNameData> | Record<string, never>
-  >;
+  getArNSRecords(
+    params?: PaginationParams,
+  ): Promise<PaginationResult<AoArNSNameData & { name: string }>>;
```

Comment:
Would be nice to have `AoArNSNameData & { name: string}` be a named type. Otherwise it bubbles out through the client's codebase and gets a bit verbose.

Reply:
[2db664a](https://github.com/ar-io/ar-io-sdk/pull/138/commits/2db664a4310837e05e25e5206f304925ec911d60)